src/replaceCustomValues.ts

```diff
@@ -174,19 +175,68 @@ export function getElementSubmissionValue({
   formatNumber,
   formatCurrency,
 }: {
-  propertyName: string
+  elementId?: string
+  propertyName?: string
   formElements: FormTypes.FormElement[]
   submission: SubmissionTypes.S3SubmissionData['submission']
 } & ReplaceInjectablesFormatters): unknown {
-  const formElement = findFormElement(
-    formElements,
-    (element) =>
-      element.type !== 'page' &&
-      element.type !== 'section' &&
-      element.name === propertyName,
-  )
+  if (elementId === undefined && propertyName === undefined) {
+    return undefined
+  }
+
+  // make sure submission is an object
+  if (Object(submission) !== submission) {
+    return undefined
+  }
+
+  let unknown: unknown = undefined
+  let formElement: FormTypes.FormElement | undefined = undefined
+
+  if (elementId) {
+    for (const element of formElements) {
+      if (elementId === element.id) {
+        if ('name' in element) {
+          unknown = submission[element.name]
+          formElement = element
+          break
+        }
+      }
+      if ('elements' in element) {
+        const newSubmissionData =
+          'name' in element ? submission[element.name] : submission
+
+        if (Array.isArray(newSubmissionData) || !element.elements) {
+          continue
+        }
+
+        const result = getElementSubmissionValue({
+          elementId,
+          formElements: element.elements,
+          //@ts-expect-error let the recursion take care of it
+          submission: newSubmissionData,
```

Comment:
Better to just cast this as `SubmissionTypes.S3SubmissionData['submission']` probably.